src/Stratis.Bitcoin.Features.Consensus/Rules/ProvenHeaderRules/ProvenHeaderCoinstakeRule.cs

```diff
@@ -0,0 +1,251 @@
+﻿using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Consensus.Rules;
+using Stratis.Bitcoin.Features.Consensus.CoinViews;
+using Stratis.Bitcoin.Features.Consensus.Interfaces;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.ProvenHeaderRules
+{
+    /// <summary>
+    /// Rule to check if the coinstake inside proven header is valid:
+    /// <list type="number">
+    ///     <item><description>Check that coinstake tx has IsCoinStake property equal to true.</description></item>
+    ///     <item><description>Header time is equal to the timestamp of the coinstake tx.</description></item>
+    ///     <item><description>Check if coinstake tx timestamp is divisible by 16 (using timestamp mask).</description></item>
+    ///     <item><description>Verify the coinstake age requirement (*).</description></item>
+    ///     <item>
+    ///     <description>
+    ///         Verify all coinstake transaction inputs (*).
+    ///         <list type="number">
+    ///             <item><description>Input comes from a UTXO.</description></item>
+    ///             <item><description>Verify the ScriptSig</description></item>
+    ///         </list>
+    ///     </description>
+    ///     </item>
+    ///     <item><description>Check if the coinstake kernel hash satisfies the difficulty requirement (*).</description></item>
+    ///     <item><description>Check that coinstake tx is in the merkle tree using the merkle proof.</description></item>
+    ///     <item><description>Verify header signature with the key from coinstake kernel (*).</description></item>
+    /// </list>
+    /// </summary>
+    /// <remarks>(*) - denotes rules that are expensive to execute.</remarks>
```

Comment:
denotes?

Reply:
Yes, that is the most appropriate word for this type of thing.